identity_iota_core/packages/iota_identity/sources/identity.move

```diff
@@ -79,12 +79,20 @@ module iota_identity::identity {
         creation_timestamp: u64,
         clock: &Clock,
         ctx: &mut TxContext
-    ): Identity {
-        let mut identity = new_with_controller(doc, ctx.sender(), false, clock, ctx);
-        assert!(identity.updated >= creation_timestamp, EInvalidTimestamp);
-        identity.created = creation_timestamp;
+    ): ID {
+        let now = clock.timestamp_ms();
+        assert!(now >= creation_timestamp, EInvalidTimestamp);
+        let identity = Identity {
+            id: object::new(ctx),
+            did_doc: multicontroller::new_with_controller(doc, ctx.sender(), false, ctx),
+            created: creation_timestamp,
+            updated: now,
+            version: PACKAGE_VERSION,
+        };
+        let id = object::id(&identity);
+        transfer::share_object(identity);
 
-        identity
+        id
     }
 
     /// Creates a new `Identity` wrapping DID DOC `doc` and controller by
```

Comment:
```suggestion
    /// Creates a new [`Identity`] wrapping DID DOC `doc` and controller by
```

Reply:
Mmm it seems _Move_ inline docs do not work the same as Rust's, using `[<ref to a piece of code>]` doesn't create a link.

Reply:
Ah, okay, just rolled with the updated version from this PR. But good to know. ^^